scripts/homework4.py

```diff
@@ -0,0 +1,125 @@
+# Author: Shad Fernandez
+# 2021-OCT-11
+# BDA 696 HW4
+
+import sys
+
+import binresponsebypredictor as bin
+import pandas as pd
+import plot_pred_response as prr
+import statsmodels.api as stats
+
+
+def main():
+    df = pd.read_csv("../datasets/Iris.csv")
+    print(df.columns)
+    response_col = str(input("Please enter the column name of the response variable: "))
+    response = [response_col]
+    predictors = df.loc[:, ~df.columns.isin(response)].columns.to_list()
+    print("Response:" + response[0])
+    print("Predictors:")
+    print(predictors)
+
+    # loop through predictors and assign if boolean or continuous
+    feature_type_dict = {}
+    for col in df.columns:
+        if df[col].dtype == "int64":
+            if df[col].unique().size == 2:
+                feature_type_dict[col] = "boolean"
+            else:
+                feature_type_dict[col] = "continuous"
+        elif df[col].dtype == "float64":
+            feature_type_dict[col] = "continuous"
+        else:
+            feature_type_dict[col] = "boolean"
+
+    # generate instance of PlotPredictorResponse
+    feature_plotter = prr.PlotPredictorResponse(df, feature_type_dict)
+    # auto generate plot based on response data type
+    feature_plotter.plot_auto(response, predictors)
+
+    # create models and fit according to dtype
+    if feature_type_dict.get(response[0]) == "continuous":
+        for pred in predictors:
+            if feature_type_dict.get(pred) == "continuous":
+                predictor = stats.add_constant(df[pred])
+                lin_reg_model_fit = stats.OLS(df[response[0]], predictor).fit()
+                print(
+                    "/nLinear regression model {} against {}".format(pred, response[0])
+                )
+                print(lin_reg_model_fit.summary())
+            else:
+                print("Please include at least one continuous predictor")
```

Comment:
Again, maybe make this its own function. Also, it looks like you may have forgot to plot the linear and/or logistic regression. It also may make sense to fish out the p and t values in order to display them more clearly then just printing out the summary of the regression models